carddav/client.go

```diff
@@ -429,3 +429,24 @@ func (c *Client) PutAddressObject(path string, card vcard.Card) (*AddressObject,
 	}
 	return ao, nil
 }
+
+// SyncCollection do a sync-collection operation on resource(path), it returns an array of modified/deleted/new
+// AddressObject, the next SyncToken and error
+func (c *Client) SyncCollection(path string, query *SyncQuery) ([]AddressObject, string, error) {
```

Comment:
Returning `[]AddressObject` isn't enough here. The server will return the new/updated resources with a 200 status, and the deleted resources with a 404. Each new/updated resource will have new/updated properties with a 200 status, and deleted properties with a 404.

We need to use a data structure capable of representing this. I think as a first version we could ignore deleted props and use something like this:

```go
type SyncResponse {
	Token string
	Updated []AddressObject
	Deleted []string
}
```

What do you think?